compiler/src/codegen/compcore.re

```diff
@@ -2047,6 +2047,94 @@ let allocate_record = (wasm_mod, env, ttag, elts) => {
   );
 };
 
+let allocate_uint_uninitialized = (wasm_mod, env, is_32_bit) => {
+  let get_swap = () => get_swap(wasm_mod, env, 0);
+  let make_alloc = () =>
+    heap_allocate(wasm_mod, env, if (is_32_bit) {2} else {4});
+
+  let (tag, label) =
+    if (is_32_bit) {
+      (Uint32Type, "allocate_unitialized_uint32");
+    } else {
+      (Uint64Type, "allocate_unitialized_uint64");
+    };
+
+  let preamble = [
+    store(
+      ~offset=0,
+      wasm_mod,
+      tee_swap(~skip_incref=true, wasm_mod, env, 0, make_alloc()),
+      Expression.Const.make(
+        wasm_mod,
+        const_int32(tag_val_of_heap_tag_type(tag)),
+      ),
+    ),
+  ];
+  let postamble = [get_swap()];
+  Expression.Block.make(
+    wasm_mod,
+    gensym_label(label),
+    List.concat([preamble, postamble]),
+  );
+};
+
+type alloc_uint_type =
+  | Uint32(Expression.t)
+  | Uint64(Expression.t);
+
+let allocate_uint = (wasm_mod, env, uint_value) => {
+  let get_swap = () => get_swap(wasm_mod, env, 0);
+
+  let (tag, instrs, needed_words, label) =
+    switch (uint_value) {
+    | Uint32(uint32) => (
+        Uint32Type,
+        [store(~offset=4, ~ty=Type.int32, wasm_mod, get_swap(), uint32)],
+        2,
+        "allocate_uint32",
+      )
+    | Uint64(uint64) => (
+        Uint64Type,
+        [store(~offset=8, ~ty=Type.int64, wasm_mod, get_swap(), uint64)],
+        // Really only 3 words needed, allocate 4 for the sake of alignment
```

Comment:
Important to call out that we _need_ 4 because of alignment:
```suggestion
        // Allocate 4 words to store with 8-byte alignment
```

Reply:
True, I think I just worded this poorly: I meant to say that since the data needs 2 words and the tag needs 1 it may be confusing why 4 words are allocated if someone didn't know about the alignment. I think your wording makes sense though